src/Nethermind/Nethermind.Db.Rocks/Config/DbConfig.cs

```diff
@@ -93,4 +94,11 @@ public class DbConfig : IDbConfig
     public bool EnableDbStatistics { get; set; } = false;
     public bool EnableMetricsUpdater { get; set; } = false;
     public uint StatsDumpPeriodSec { get; set; } = 600;
+
+    public int DisableAutoCompactions { get; set; } = 1;
```

Comment:
No, no we can't set this to true by default. Read this https://github.com/EighteenZi/rocksdb_wiki/blob/master/RocksDB-Tuning-Guide.md#user-content-level-style-compaction.

Reply:
Could you please elaborate and explain why exactly?

Reply:
The term `compaction` here does not really compact anything for us as we don't delete anything or modify existing key. The `compaction` here is also used to mantain the structure of the db. If it is disable, the L0 file count will increase without bound, normally its probably around 4 to 8, less it better. In order to fetch a key, rocksdb will:
1. Read memtable, if not found,
2. Read ALL L0 file (L0 file is special), if not found,
3. Read One file from L1 to LN (where N is probably 4 in our case).

Note, bloom filter will make checking file very fast, but it still need to check them. Notice, L0 is special as the key range between file overlap other file in L0. When it compact L0 to L1, Different range of key will go to different file in L1, hence, a key will always be in a single file in L1.

Reply:
I still do not see what is the problem. Please explain what are the potential/expected problems?

Reply:
As the number of L0 file increase, each read will need to go through an increasing number of file. Now, because you tested it with pruning turned on, chances are, its just the blocks db that is causing the compaction spike, because blocks are large, and its memory budget is only 8 MB, so each write will probably trigger flush, and it will probably trigger a compaction every 4-8 block. Blocks don't need low read access time, which is probably the reason why you don't see much change. But state does. Try running it with pruning off, and check the overall (average) block processing time. It may be fine for one day, but after one week the block processing time will increase.

Reply:
Please help us compose a definitive synthetic test for our use-case that can test your expectations. We really need to work with something measurable and repeatable to continue this discussion in a constructive way.